tests/test_updater.py

```diff
@@ -0,0 +1,253 @@
+
+
+def test_pre_0_4_2_settings(tmp_path: Path, monkeypatch: MonkeyPatch) -> None:
+    """Check settings of installations prior to 0.4.2.
+
+    Check that installations that have been upgraded from a version < 0.4.2 to >= 0.4.2
+    will automatically get the default updater settings, even though they never existed
+    in their settings.json file.
+    """
+    save_settings(tmp_path, default_settings_0_4_1())
+    updater = generate_isolated_updater(tmp_path, monkeypatch)
+    assert updater.dangerzone.settings.get("updater") == default_updater_settings()
+
+
+def test_post_0_4_2_settings(tmp_path: Path, monkeypatch: MonkeyPatch) -> None:
+    """Check settings of installations post-0.4.2.
+
+    Installations from 0.4.2 onwards will have a "latest_version" field in their updater
+    settings. When these installations get upgraded to a newer version, we must make
+    sure that this fields becomes equal to the new version, so that the user is not
+    erroneously prompted to a version they already have.
+    """
+    # Store the settings of Dangerzone 0.4.2 to the filesystem.
+    old_settings = settings.Settings.generate_default_settings()
+    old_settings["updater"]["latest_version"] == "0.4.2"
+    save_settings(tmp_path, old_settings)
+
+    # Mimic an upgrade to version 0.4.3, by making Dangerzone report that the current
+    # version is 0.4.3.
+    expected_settings = default_updater_settings()
+    expected_settings["latest_version"] = "0.4.3"
+    monkeypatch.setattr(
+        settings, "get_version", lambda: expected_settings["latest_version"]
+    )
+
+    # Ensure that the Settings class will correct the latest version field to 0.4.3.
+    updater = generate_isolated_updater(tmp_path, monkeypatch)
+    assert updater.dangerzone.settings.get("updater") == expected_settings
+
+    # Simulate an updater check that found a newer Dangerzone version (e.g., 0.4.4).
+    expected_settings["latest_version"] = "0.4.4"
+    updater.dangerzone.settings.updater_set(
+        "latest_version", expected_settings["latest_version"]
+    )
+    updater.dangerzone.settings.save()
+
+    # Ensure that the Settings class will leave the "latest_version" field intact the
+    # next time we reload the settings.
+    updater.dangerzone.settings.load()
+    assert updater.dangerzone.settings.get("updater") == expected_settings
+
+
+# FIXME: This test should not fail
+@pytest.mark.xfail(reason="We haven't enabled this behavior in Linux yet")
+def test_linux_first_run(updater: Updater) -> None:
+    """Check that Linux users are not prompted for update checks."""
+    expected_settings = default_updater_settings()
+    expected_settings["first_run"] = False
+    expected_settings["check"] = False
+    assert updater.should_check_for_updates() == False
+    assert updater.dangerzone.settings.get("updater") == expected_settings
+
+
+# FIXME: Make this check run only on Windows/MacOS platforms.
```

Comment:
Right, but for dev purposes, these checks still run on Linux. I'll use the snippet you provided once we fix this:

https://github.com/freedomofpress/dangerzone/blob/e6cd583946e26d09c829afd286b4b2ce3a8f9f48/dangerzone/gui/updater.py#L109-L114

Reply:
Note that this is now irrelevant, since we have a way to both test the update logic on Linux and test that Linux users will not see update notifications.